src/MessageWriter.ts

```diff
@@ -213,7 +232,7 @@ export class MessageWriter {
         } else {
           // Primitive array
           const arrayWriter = this.getPrimitiveArrayWriter(field.type);
-          arrayWriter(nestedMessage, field.defaultValue, writer);
+          arrayWriter(nestedMessage, field.defaultValue, writer, field.arrayLength);
```

Comment:
The following would have also worked without changing the array writer signature, but then typings wouldn't be correct anymore.
```suggestion
          arrayWriter(nestedMessage, field.defaultValue ?? new Array(field.arrayLength), writer);
```